src/react/honeypot-inputs.tsx

```diff
@@ -1,51 +1,58 @@
 import * as React from "react";
-import { useRouteLoaderData } from "@remix-run/react";
-import type { HoneypotInputProps } from "../server/honeypot.jsx";
+import { type HoneypotInputProps } from "../server/honeypot.js";
 
-export function HoneypotInputs() {
-  let rootLoaderData = useRouteLoaderData(
-    "root"
-  ) as Partial<HoneypotInputProps>;
```

Comment:
Pulling in the root loader data is magic and especially since the properties are generic like "nameFieldName" that could be very confusing.

Changing to context (a la the CSRF token) puts the control back into the hands of the developer and is much nicer to work with IMO.